cluster/examples/kubernetes/ceph/create-external-cluster-resources.py

```diff
@@ -528,9 +560,61 @@ def gen_json_out(self):
             })
         return json.dumps(json_out)+LINESEP
 
+    def upgrade_user_permissions(self):
+        # check whether the given user exists or not
+        cmd_json = {"prefix": "auth get", "entity": self.run_as_user, "format": "json"}
+        ret_val, json_out, err_msg = self._common_cmd_json_gen(cmd_json)
+        if ret_val != 0 or len(json_out) == 0:
+            raise ExecutionFailureException("'auth get {}' command failed.\n".format(self.run_as_user) +
+                "Error: {}".format(err_msg if ret_val != 0 else self.EMPTY_OUTPUT_LIST))
+        j_first = json_out[0]
+        existing_caps = j_first['caps']
+        osd_cap = "osd"
+        cap_keys = ["mon", "mgr", "osd"]
+        for eachCap in cap_keys:
+            min_cap_values = self.MIN_USER_CAP_PERMISSIONS.get(eachCap, '')
+            cur_cap_values = existing_caps.get(eachCap, '')
+            # detect rgw-pool-prefix
+            if eachCap == osd_cap:
+                # if directly provided through '--rgw-pool-prefix' argument, use it
+                if self._arg_parser.rgw_pool_prefix:
+                    min_cap_values = min_cap_values.format(self._arg_parser.rgw_pool_prefix)
+                # or else try to detect one from the existing/current osd cap values
+                else:
+                    rc = re.compile(r' pool=(\w+)\.rgw\.[^ ]*')
+                    # 'findall()' method will give a list of prefixes
+                    # and 'set' will eliminate any duplicates
+                    cur_rgw_pool_prefix_list = list(set(rc.findall(cur_cap_values)))
+                    if len(cur_rgw_pool_prefix_list) < 1:
+                        raise ExecutionFailureException("Unable to detect an existing 'rgw-pool-prefx'. Please provide one with '--rgw-pool-prefix'")
+                    elif len(cur_rgw_pool_prefix_list) > 1:
+                        raise ExecutionFailureException("Multiple 'rgw_pool_prefix' found, {}. Please provide an exact pool prefix using '--rgw-pool-prefix' argument".format(cur_rgw_pool_prefix_list))
```

Comment:
I don't think this is possible unless the keycaps were altered by the admin... In that case, we should not do anything.